lib/multi.c

```diff
@@ -1361,9 +1362,11 @@ static CURLMcode multi_wait(struct Curl_multi *multi,
           if(ret && pollrc <= 0 && wsa_events.lNetworkEvents)
             retcode++;
         }
-        WSAEventSelect(extra_fds[i].fd, multi->wsa_event, 0);
-        if(pollrc <= 0)
+        WSAEventSelect(s, multi->wsa_event, 0);
+        if(pollrc <= 0) {
+          extra_fds[i].revents = mask;
```

Comment:
Maybe get rid of the `mask` variable and always use `extra_fds[i].revents` instead to avoid this mistake completely?

Reply:
I think the mask variable is there to avoid redundant array access in this performance relevant code path.

Reply:
Should this still ignore Curl_poll() returning -1 ? We fixed it elsewhere previously in 5912da253

Reply:
> I think the mask variable is there to avoid redundant array access in this performance relevant code path.

I don't see anything wrong with it but I doubt we are saving many cycles here.

> Should this still ignore Curl_poll() returning -1 ? We fixed it elsewhere previously in [5912da2](https://github.com/curl/curl/commit/5912da253b64de3cb2a1a229558077219b2c8a35)

It seems that fix covered this code as well, it's just not visible in the review unless you use the up arrow to expand more lines. Though further up there is a place where it could happen,

https://github.com/curl/curl/blob/45ac4d019475df03562fe0ac54eb67e1d1de0ca7/lib/multi.c#L1307-L1314

Reply:
I will look into this in a separate follow-up PR.

Reply:
> Should this still ignore Curl_poll() returning -1 ? We fixed it elsewhere previously in [5912da2](https://github.com/curl/curl/commit/5912da253b64de3cb2a1a229558077219b2c8a35)

@bagder I believe https://github.com/curl/curl/pull/9372 could be a good attempt at improving this situation here.